src/nbl/video/CVulkanRayTracingPipeline.cpp

```diff
@@ -0,0 +1,61 @@
+#include "nbl/video/CVulkanRayTracingPipeline.h"
+
+#include "nbl/video/CVulkanLogicalDevice.h"
+
+namespace nbl::video
+{
+
+  CVulkanRayTracingPipeline::CVulkanRayTracingPipeline(const SCreationParams& params, const VkPipeline vk_pipeline) :
+    IGPURayTracingPipeline(params),
+    m_vkPipeline(vk_pipeline),
+    m_shaders(core::make_refctd_dynamic_array<ShaderContainer>(params.shaders.size()))
+  {
+    for (size_t shaderIx = 0; shaderIx < params.shaders.size(); shaderIx++)
+      m_shaders->operator[](shaderIx) = ShaderRef(static_cast<const CVulkanShader*>(params.shaders[shaderIx].shader));
+
+    const auto* vulkanDevice = static_cast<const CVulkanLogicalDevice*>(getOriginDevice());
+    const auto handleCount = params.shaderGroups.getShaderGroupCount();
+    const auto handleSize = SPhysicalDeviceLimits::ShaderGroupHandleSize;
+    const auto dataSize = handleCount * handleSize;
+    auto* vk = vulkanDevice->getFunctionTable();
+    m_shaderGroupHandles = core::make_refctd_dynamic_array<ShaderHandleContainer>(dataSize);
+    vk->vk.vkGetRayTracingShaderGroupHandlesKHR(vulkanDevice->getInternalObject(), m_vkPipeline, 0, handleCount, dataSize, m_shaderGroupHandles->data());
```

Comment:
can this function call fail with a `VK_ERROR_...` of any sort?

If yes, then needs to be done in 
`CVulkanLogicalDevice::createRayTracingPipelines_impl` before entering the constructor here.

Reply:
ok I checked, only OOM errors can be returned (which we dont generally handle) so can stay here.